agent/engine/housing.go

```diff
@@ -80,15 +82,15 @@ func maybeAdd(original []string, add string) []string {
 }
 
 func occupyStart(inner dependency.StartFunc, name string) dependency.StartFunc {
-	return func(context dependency.Context) (worker.Worker, error) {
+	return func(ctx context.Context, getter dependency.Getter) (worker.Worker, error) {
 		var guest fortress.Guest
-		if err := context.Get(name, &guest); err != nil {
+		if err := getter.Get(name, &guest); err != nil {
 			return nil, errors.Trace(err)
 		}
 		task := func() (worker.Worker, error) {
-			return inner(context)
+			return inner(ctx, getter)
 		}
-		worker, err := fortress.Occupy(guest, task, context.Abort())
+		worker, err := fortress.Occupy(guest, task, ctx.Done())
```

Comment:
I'm sure the std `ctx.Done()` here is functionally equivalent to the old`context.Abort()`.

Reply:
In this context it does, I had to change the `worker.Worker` package to ensure it was.